File: tests/Generator/resource/php/oop/RootSchema.php
<?php

declare(strict_types = 1);

class RootSchema implements \JsonSerializable, \PSX\Record\RecordableInterface
{
    protected ?StudentMap $students = null;
    public function setStudents(?StudentMap $students) : void
    {
        $this->students = $students;
    }
    public function getStudents() : ?StudentMap
    {
        return $this->students;
    }
    public function toRecord() : \PSX\Record\RecordInterface
    {
        /** @var \PSX\Record\Record<mixed> $record */
        $record = new \PSX\Record\Record();
        $record->put('students', $this->students);
        return $record;
    }
    public function jsonSerialize() : object
    {
        return (object) $this->toRecord()->getAll();
    }
}

File: tests/Generator/resource/java/oop/RootSchema.java

import com.fasterxml.jackson.annotation.*;

public class RootSchema {
    private StudentMap students;

    @JsonSetter("students")
    public void setStudents(StudentMap students) {
        this.students = students;
    }

    @JsonGetter("students")
    public StudentMap getStudents() {
        return this.students;
    }
}

